src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -7680,9 +7680,13 @@ void StorageReplicatedMergeTree::forcefullyRemoveBrokenOutdatedPartFromZooKeeper
     /// It could became Outdated because it was merged/mutated (so we have a covering part) or because it was dropped.
     /// But DROP [PART]ITION waits for all Outdated parts to be loaded, so it's not the case.
 
-    auto zookeeper = getZooKeeper();
+    bool exists = false;
     String part_path = replica_path + "/parts/" + part_name;
-    if (!zookeeper->exists(part_path))
+    ZooKeeperRetriesInfo retries_info{/* max_retries */5, /* initial_backoff_ms */100, /* max_backoff_ms */100};
```

Comment:
Not sure if a new setting needs to be introduced for these, currently there are settings for keeper retries in backups and in inserts only.

Reply:
I would use the insert settings, we did the same for system.zookeeper. We should probably unify both insert and backup settings and have a universal set of settings that works for any query.